examples/random/src/main.cpp

```diff
@@ -20,115 +20,132 @@
+    for (int i = 0; i < 5; i++)
+    {
+        std::cout << x.begin()[i] << std::endl;
+    }
 
-  std::int64_t nsamples = 100;
-  constexpr int vec_size = 4;
+    std::cout << "\nLast 5 samples of minstd_rand with scalar generation" << std::endl;
+    for (int i = 0; i < 5; i++)
+    {
+        std::cout << x.rbegin()[i] << std::endl;
+    }
+}
 
-  std::vector<float> x(nsamples);
+template <int VecSize, typename RealType>
+void
+vector_example(sycl::queue& queue, const ::std::uint32_t seed, std::vector<RealType>& x)
+{
+    {
+        sycl::buffer<RealType, 1> x_buffer(x.data(), sycl::range<1>(x.size()));
+
+        queue.submit(
+            [&](sycl::handler& cgh)
+            {
+                auto x_acc = x_buffer.template get_access<sycl::access::mode::write>(cgh);
+
+                cgh.parallel_for<class generate_kernel>(
+                    sycl::range<1>(x.size() / VecSize),
+                    [=](sycl::item<1> idx)
+                    {
+                        unsigned long long offset = idx.get_linear_id() * VecSize;
+
+                        // Create minstd_rand engine
+                        oneapi::dpl::minstd_rand_vec<VecSize> engine(seed);
+                        engine.discard(offset);
+
+                        // Create float uniform_real_distribution distribution
+                        oneapi::dpl::uniform_real_distribution<sycl::vec<float, VecSize>> distr;
+
+                        // Generate sycl::vec<float, VecSize> of random numbers
+                        auto res = distr(engine);
+
+                        // Store results from res to VecSize * offset position of x_acc
+                        res.store(idx.get_linear_id(), x_acc.get_pointer());
+                    });
+            });
+    }
 
-  // Scalar random number generation
-  scalar_example(queue, x);
+    std::cout << "\nsuccess for vector generation" << std::endl;
+    std::cout << "First 5 samples of minstd_rand with vector generation" << std::endl;
+    for (int i = 0; i < 5; i++)
+    {
+        std::cout << x.begin()[i] << std::endl;
+    }
 
-  // Vector random number generation
-  vector_example<vec_size>(queue, x);
+    std::cout << "\nLast 5 samples of minstd_rand with vector generation" << std::endl;
+    for (int i = 0; i < 5; i++)
+    {
+        std::cout << x.rbegin()[i] << std::endl;
+    }
+}
 
-  return 0;
-}
\ No newline at end of file
+int
+main()
+{
+    auto async_handler = [](sycl::exception_list ex_list)
+    {
+        for (auto& ex : ex_list)
+        {
+            try
+            {
+                std::rethrow_exception(ex);
+            }
+            catch (sycl::exception& ex)
+            {
+                std::cerr << ex.what() << std::endl;
+                std::exit(1);
+            }
+        }
+    };
+
+    sycl::queue queue(sycl::default_selector{}, async_handler);
+
+    std::int64_t nsamples = 100;
+    constexpr int vec_size = 4;
+    constexpr ::std::uint32_t seed = 777;
```

Comment:
everthing in this file is written with ``std::`` and only this one is written with `::std::``
```suggestion
    constexpr std::uint32_t seed = 777;
```

Reply:
Fixed